### PR TITLE
*: fix the prepare checker data race

### LINKED ISSUE #4987: Data race in
### Which jobs are failing

FAIL	github.com/tikv/pd/tests/server/cluster	98.230s

### CI link

<https://github.com/tikv/pd/runs/6501064606?check_suite_focus=true>

### Reason for failure (if possible)

```
WARNING: DATA RACE
Write at 0x00c0033313a0 by goroutine 64:
  github.com/tikv/pd/server/cluster.(*prepareChecker).check()
      /home/runner/work/pd/pd/server/cluster/prepare_checker.go:64 +0x545
  github.com/tikv/pd/server/cluster.(*coordinator).shouldRun()
      /home/runner/work/pd/pd/server/cluster/coordinator.go:624 +0x12d
  github.com/tikv/pd/server/cluster.(*coordinator).run()
      /home/runner/work/pd/pd/server/cluster/coordinator.go:320 +0x1b8
  github.com/tikv/pd/server/cluster.(*coordinator).runUntilStop()
      /home/runner/work/pd/pd/server/cluster/coordinator.go:305 +0x44
  github.com/tikv/pd/server/cluster.(*RaftCluster).runCoordinator()
      /home/runner/work/pd/pd/server/cluster/cluster.go:437 +0xdd
  github.com/tikv/pd/server/cluster.(*RaftCluster).Start.func2()
      /home/runner/work/pd/pd/server/cluster/cluster.go:271 +0x39

Previous read at 0x00c0033313a0 by goroutine 744:
  github.com/tikv/pd/server/cluster.(*prepareChecker).isPrepared()
      /home/runner/work/pd/pd/server/cluster/prepare_checker.go:80 +0x119
  github.com/tikv/pd/server/cluster.(*RaftCluster).IsPrepared()
      /home/runner/work/pd/pd/server/cluster/cluster.go:758 +0x74
  github.com/tikv/pd/server/cluster.(*RaftCluster).processRegionHeartbeat()
      /home/runner/work/pd/pd/server/cluster/cluster.go:827 +0xf19
  github.com/tikv/pd/server/cluster.(*RaftCluster).HandleRegionHeartbeat()
      /home/runner/work/pd/pd/server/cluster/cluster_worker.go:35 +0x4b
  github.com/tikv/pd/tests/server/cluster_test.(*clusterTestSuite).TestConcurrentHandleRegion.func2()
      /home/runner/work/pd/pd/tests/server/cluster/cluster_test.go:657 +0x119

Goroutine 64 (running) created at:
  github.com/tikv/pd/server/cluster.(*RaftCluster).Start()
      /home/runner/work/pd/pd/server/cluster/cluster.go:271 +0xab9
  github.com/tikv/pd/server.(*Server).bootstrapCluster()
      /home/runner/work/pd/pd/server/server.go:674 +0x17b1
  github.com/tikv/pd/server.(*GrpcServer).Bootstrap()
      /home/runner/work/pd/pd/server/grpc_service.go:401 +0x43e
  github.com/pingcap/kvproto/pkg/pdpb._PD_Bootstrap_Handler.func1()
      /home/runner/go/pkg/mod/github.com/pingcap/kvproto@v0.0.0-20220510035547-0e2f26c0a46a/pkg/pdpb/pdpb.pb.go:7359 +0x88
  github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1()
      /home/runner/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.0.1-0.20190118093823-f849b5445de4/chain.go:31 +0x238
  github.com/grpc-ecosystem/go-grpc-prometheus.(*ServerMetrics).UnaryServerInterceptor.func1()
      /home/runner/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-prometheus@v1.2.0/server_metrics.go:107 +0xc7
  github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1()
      /home/runner/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.0.1-0.20190118093823-f849b5445de4/chain.go:34 +0x18c
  go.etcd.io/etcd/etcdserver/api/v3rpc.newUnaryInterceptor.func1()
      /home/runner/go/pkg/mod/go.etcd.io/etcd@v0.5.0-alpha.5.0.20191023171146-3cf2f69b5738/etcdserver/api/v3rpc/interceptor.go:64 +0x2e6
  github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1()
      /home/runner/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.0.1-0.20190118093823-f849b5445de4/chain.go:34 +0x18c
  go.etcd.io/etcd/etcdserver/api/v3rpc.newLogUnaryInterceptor.func1()
      /home/runner/go/pkg/mod/go.etcd.io/etcd@v0.5.0-alpha.5.0.20191023171146-3cf2f69b5738/etcdserver/api/v3rpc/interceptor.go:71 +0xc7
  github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1()
      /home/runner/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.0.1-0.20190118093823-f849b5445de4/chain.go:39 +0x35b
  github.com/pingcap/kvproto/pkg/pdpb._PD_Bootstrap_Handler()
      /home/runner/go/pkg/mod/github.com/pingcap/kvproto@v0.0.0-20220510035547-0e2f26c0a46a/pkg/pdpb/pdpb.pb.go:7361 +0x1da
  google.golang.org/grpc.(*Server).processUnaryRPC()
      /home/runner/go/pkg/mod/google.golang.org/grpc@v1.26.0/server.go:1024 +0x1471
  google.golang.org/grpc.(*Server).handleStream()
      /home/runner/go/pkg/mod/google.golang.org/grpc@v1.26.0/server.go:1313 +0xf53
  google.golang.org/grpc.(*Server).serveStreams.func1.1()
      /home/runner/go/pkg/mod/google.golang.org/grpc@v1.26.0/server.go:722 +0xec

Goroutine 744 (running) created at:
  github.com/tikv/pd/tests/server/cluster_test.(*clusterTestSuite).TestConcurrentHandleRegion()
      /home/runner/work/pd/pd/tests/server/cluster/cluster_test.go:655 +0x1491
  runtime.call16()
      /opt/hostedtoolcache/go/1.18.0/x64/src/runtime/asm_amd64.s:701 +0x48
  reflect.Value.Call()
      /opt/hostedtoolcache/go/1.18.0/x64/src/reflect/value.go:339 +0xd7
  github.com/pingcap/check.(*suiteRunner).forkTest.func1()
      /home/runner/go/pkg/mod/github.com/pingcap/check@v0.0.0-20[211](https://github.com/tikv/pd/runs/6501064606?check_suite_focus=true#step:5:212)0261[254](https://github.com/tikv/pd/runs/6501064606?check_suite_focus=true#step:5:255)17-57bd13f7b5f0/check.go:850 +0xa71
  github.com/pingcap/check.(*suiteRunner).forkCall.func1()
      /home/runner/go/pkg/mod/github.com/pingcap/check@v0.0.0-202110[261](https://github.com/tikv/pd/runs/6501064606?check_suite_focus=true#step:5:262)25417-57bd13f7b5f0/check.go:739 +0x11e
```

### PR DESCRIPTION
**Pull request by rleungx**
Signed-off-by: Ryan Leung <rleungx@gmail.com>

### What problem does this PR solve?

Issue Number: Close #4987.

### What is changed and how does it work?
This PR fixes the problem by replacing `Rlock` with `Lock`.

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None
```

### CODE REVIEW

File: server/cluster/prepare_checker.go [hunk 1]
```diff
@@ -38,8 +38,8 @@ func newPrepareChecker() *prepareChecker {
 
 // Before starting up the scheduler, we need to take the proportion of the regions on each store into consideration.
 func (checker *prepareChecker) check(c *core.BasicCluster) bool {
-	checker.RLock()
-	defer checker.RUnlock()
+	checker.Lock()
```
Comment (lhy1024) [inline server/cluster/prepare_checker.go:41]: line64 `checker.prepared = true`

**Review by nolouch — APPROVED** [review]

File: server/cluster/prepare_checker.go (hunk 1 above)
Comment (rleungx) [reply server/cluster/prepare_checker.go:41]: What do you mean?

**Review by lhy1024 — APPROVED** [review]

**Pull request merged**